aws_lambda_builders/workflows/python_pip/workflow.py

```diff
@@ -92,8 +92,16 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
                 self.actions.append(
                     CopySourceAction(self.dependencies_dir, artifacts_dir, excludes=self.EXCLUDED_FILES)
                 )
+
         else:
             LOG.warning("requirements.txt file not found. Continuing the build without dependencies.")
+
+        if not self.download_dependencies and not self.dependencies_dir:
+            LOG.info(
+                "download_dependencies is false and the dependencies_dir is not exists, just copying the source files "
+                "into the artifacts directory "
+            )
```

Comment:
I would move validation to the above so that we will fail or emit this message early.